### DISCUSSION TITLE
Is the only way to change the wording in `pagy_info` to monkey patch the gem?

### DISCUSSION THREAD

**Discussion by DmitrySychev** [ddnexus/pagy#489] [Q&A] [answered]
Primarily, the method only allows you to customize the 'items' name. I'd love to be able to change the wording entirely, meaning no "Displaying items...". 

I see that it's listed in the gems locales, but there doesn't seem to be a way to access from outside the gem.

**Comment by ddnexus** [comment]
> I see that it's listed in the gems locales, but there doesn't seem to be a way to access from outside the gem.

Customizing the dictionary IS by definition the "way to access [and edit the static strings] outside the gem".

**Reply by DmitrySychev** [reply]
Fair enough, I think I can explain it better. The variable that the gem allows a user to change externally is the `items` word out of the whole sentence. That is the only argument you are able to pass to the method, the rest of the sentence is defined in the locales that are internal to the gem and are not changeable without monkey patching. 

See `single_page` and `multiple_pages`:
https://github.com/ddnexus/pagy/blob/master/lib/locales/en.yml

In practice, this is what you'd see in your view `Displaying items 1-5 of 50 in total`, and ideally, I'd love to be able to customize it to just `1-5 of 50`

If I understand the usage correctly, that's what I see.

**Reply by ddnexus** [reply]
You have just to customize the dictionary files for i18n by removing what you don't want to display from all the entries in the dictionary. Pagy will take your own custom strings to compose its output.

For `mutiple_pages` for example:

```
multiple_pages: "%{from}-%{to} of %{count}"
```
will give you what you want without the need of monkey-patching anything.

**Reply by ddnexus** [reply]
BTW, customizing the dictionary depends on what you use as i18n. For pure pagy look [here](https://ddnexus.github.io/pagy/docs/api/i18n/#pagyi18n). If you use the standard `i18n` gem, then research how to override entries in i18n.

(We have issue #452 to improve the doc about how to do that.)

**Reply by DmitrySychev** [reply]
That's awesome. I haven't worked with dictionaries and locales before, but it makes sense that you can overwrite those values. Appreciate the insight. Going to read up on this. Thank you!

**Comment by ddnexus** [comment, marked as answer]
See [How to customize the dictionary](https://ddnexus.github.io/pagy/docs/how-to/#customize-the-dictionary)
